packages/manager/modules/pci/src/projects/project/public-gateways/add/add.controller.js

```diff
@@ -0,0 +1,225 @@
+      : {
+          id: '',
+          name: this.$translate.instant(transKey),
+        };
+
+  onGatewaySizeFocus() {
+    this.displaySelectedGateway = false;
+  }
+
+  onGatewaySizeChange() {
+    this.trackPublicGateways('add::public-gateway_add_select-type');
+    this.displaySelectedGateway = true;
+  }
+
+  onRegionsFocus() {
+    this.displaySelectedRegion = false;
+  }
+
+  onRegionChange(region) {
+    this.trackPublicGateways('add::public-gateway_add_select-region');
+    this.displaySelectedRegion = true;
+    this.selectedRegion = region;
+    this.getSelectedRegionNetwork(this.projectId, this.selectedRegion.name);
+  }
+
+  getSelectedRegionNetwork(projectId, regionName) {
+    this.PciPublicGatewaysService.getPrivateNetworks(projectId, regionName)
+      .then((network) => {
+        this.privateNetworks = network;
+      })
+      .catch((error) => this.CucCloudMessage.error(get(error, 'data.message')));
+  }
+
+  onGatewaySizeSelect(gateway) {
+    this.selectedGatewaySize = gateway;
+  }
+
+  onAddPrivateNetworkClick() {
+    this.trackPublicGateways('add::add-private-network');
+    this.showAddPrivateNetworkModalForm = true;
+  }
+
+  onCancel() {
+    this.trackPublicGateways('add::add-private-network::cancel');
+    this.showAddPrivateNetworkModalForm = false;
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(
+      'pci.projects.project.public-gateways.add',
+    );
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      'pci.projects.project.public-gateways.add',
+      {
+        onMessage: () => this.refreshMessages(),
+      },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getGatewayModel = (gatewaySize) => gatewaySize.split(/[-]+/).pop();
+
+  getNetworkSubnet(projectId, regionName, networkId) {
+    this.PciPublicGatewaysService.getSubnet(projectId, regionName, networkId)
+      .then((subnet) => {
+        this.networkSubnet = subnet;
+      })
+      .catch((error) => this.CucCloudMessage.error(get(error, 'data.message')));
+  }
+
+  onPrivateNetworkChange(selectedNetwork) {
+    this.trackPublicGateways('add::public-gateway_add_add-private-network');
+    this.getNetworkSubnet(
+      this.projectId,
+      this.selectedRegion.name,
+      selectedNetwork.id,
+    );
+  }
+
+  onAddPrivateNetworkFormSubmit() {
+    this.trackPublicGateways('add::add-private-network::confirm');
+    this.isLoading = true;
+    this.privateNetworkModel = {
+      gateway: {
+        name: this.gatewayName,
+        model: this.getGatewayModel(this.selectedGatewaySize.product),
+      },
+      name: this.privateNetworkModel.name,
+      subnet: {
+        cidr: this.selectedSubnet,
+        ipVersion: DEFAULT_IPVERSION,
+      },
+    };
+    this.PciPublicGatewaysService.addPrivateNetwork(
+      this.projectId,
+      this.selectedRegion.name,
+      this.privateNetworkModel,
+    )
+      .then((data) => {
+        this.isLoading = false;
```

Comment:
**nitpick(non-blocking):** Common statements in the `.then()` and `.catch()` can be moved to `.finally()`